Lib/doctest.py

```diff
@@ -235,25 +236,32 @@ def _normalize_module(module, depth=2):
     else:
         raise TypeError("Expected a module, string, or None")
 
-def _newline_convert(data):
-    # The IO module provides a handy decoder for universal newline conversion
-    return IncrementalNewlineDecoder(None, True).decode(data, True)
-
 def _load_testfile(filename, package, module_relative, encoding):
     if module_relative:
         package = _normalize_module(package, 3)
+        try:
+            loader = package.__loader__
+        except AttributeError:
+            pass
+        else:
+            if loader is not None:
+                return (
+                    importlib.resources.read_text(package, filename, encoding=encoding),
+                    filename,
+                )
+
+        try:
+            package.__spec__.loader
+        except AttributeError:
+            pass
+        else:
+            return (
+                importlib.resources.read_text(package, filename, encoding=encoding),
+                filename,
+            )
```

Comment:
I'm thinking the duplication here is unnecessary. In fact, I can't tell from reading the code why there are two paths that are checked before invoking the same behavior. Why not instead:

```suggestion
        with contextlib.suppress(Exception):
            return (
                importlib.resources.read_text(package, filename, encoding=encoding),
                filename,
            )
```

Reply:
My feeling is that doctest shouldn't be concerning itself with the implementation details of whether a module should have loadable resources or not, but should be able to rely on `importlib.resources` to assert that state.

In https://github.com/jaraco/cpython/tree/gh-128862/importlib-resources-load, I've first refactored the code to attempt to disentangle concerns. I've then trapped `AttributeError` (what importlib.resources happens to throw) when loading the resource. I considered trapping `Exception` here, but I'm worried that might mask other unexpected conditions.

Even better would be if `importlib.resources` could throw a more specific exception when the indicated module is unable to resolve resources, but since it currently just raises an AttributeError, perhaps it's best to just trap that exception, rather than try to pre-emptively detect that exception will occur.